build(aio): various npm upgrades

IgorMinar opened the pull request.

petebacondarwin reviewed (approved):
Assuming that Travis is happy, I am happy.
No breaking issues in any of these upgrades??

IgorMinar commented:
@petebacondarwin I had to lower the pwa score since we got down to 89. I don't know why. I tried to compare before and after and didn't see any meaningful difference.

@gkalpak I wonder how to make debugging the lighthouse result easier. any ideas. should we store the last report somewhere so that we can compare what changed? or is the output too noisy and unstable?

mary-poppins commented:
The angular.io preview for bdedf25e5fe34d11897334e6fab760010d3de483 is available [here][1].

[1]: https://pr16665-bdedf25e5fe34d11897334e6fab760010d3de483.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 50319b3aae6a6b5eb1e1d5bcbb31b3b2da2e75f4 is available [here][1].

[1]: https://pr16665-50319b3aae6a6b5eb1e1d5bcbb31b3b2da2e75f4.ngbuilds.io/

gkalpak commented on aio/package.json:
```diff
@@ -44,18 +44,18 @@
     "@angular/platform-browser-dynamic": "next",
-    "@angular/platform-server": "next",
-    "@angular/router": "next",
+    "@angular/router": "^4.1.1",
```
Why `^4.1.1` for router and `next` for everything else?

IgorMinar replied:
ah. that's a mistake. I removed it and then cli failed because it has a hard dependency on the router. so I had to readd it. I'll fix it.

gkalpak commented on aio/package.json:
```diff
@@ -15,7 +15,7 @@
     "setup": "yarn && yarn boilerplate:add && yarn generate-plunkers && yarn generate-zips && yarn docs",
     "pretest-pwa-score-local": "yarn build",
-    "test-pwa-score-local": "concurrently --kill-others --success first \"http-server dist -p 4200 --silent\" \"yarn test-pwa-score -- http://localhost:4200 90\"",
+    "test-pwa-score-local": "concurrently --kill-others --success first \"http-server dist -p 4200 --silent\" \"yarn test-pwa-score -- http://localhost:4200 88\"",
```
This will only lower the target score for local testing (against the locally built app).
For lowering the target score for the test against the deployed version, you need to update [env.sh][1].

[1]: https://github.com/IgorMinar/angular/blob/50319b3aae6a6b5eb1e1d5bcbb31b3b2da2e75f4/scripts/ci/env.sh#L64

IgorMinar replied:
sreally? I thought we agreed to have this number only in one place.

gkalpak replied:
Well...the idea with `test-pwa-score-local` was to be able to run it locally (without having to source `env.sh` which contains the vars. We are running this in CI as well as a smoke-test, so we fail early (and not after depoying the preview) if we did something horribly wrong wrt PWA.

I can add a third value in `env.sh` (for CI) and pass it manually when running this locally if you want me to.

gkalpak commented:
We could store the results (but they are easy to obtain if there is a preview).

The only area that we lose "points" is in "Page load performance is fast" (especially subsections [First meaningful paint][1] and [Time To Interactive (alpha)][2]). The reports are somewhat noisy (I typically results varying by +/-2 between runs), but have been fairly stable on Travis (+/-0.5).

A drop from 93 (previous score) to 89 is quite significant though. It doesn't seem like flake 😕 

[1]: https://developers.google.com/web/tools/lighthouse/audits/first-meaningful-paint
[2]: https://developers.google.com/web/tools/lighthouse/audits/time-to-interactive

mary-poppins commented:
The angular.io preview for b911b628eb87ea72fd37f032fa66761b72400275 is available [here][1].

[1]: https://pr16665-b911b628eb87ea72fd37f032fa66761b72400275.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 354debe89e6e4c23d2683c1bb4cdfdfd8fecfed7 is available [here][1].

[1]: https://pr16665-354debe89e6e4c23d2683c1bb4cdfdfd8fecfed7.ngbuilds.io/

mary-poppins commented:
The angular.io preview for c9bf2517757da4f70627230f576811efb45f0a63 is available [here][1].

[1]: https://pr16665-c9bf2517757da4f70627230f576811efb45f0a63.ngbuilds.io/

mary-poppins commented:
The angular.io preview for df465078f16425c1906d4e14b3ba52c9fdbb98f0 is available [here][1].

[1]: https://pr16665-df465078f16425c1906d4e14b3ba52c9fdbb98f0.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 3dba7f732330ad64955bbac009a428331c08d382 is available [here][1].

[1]: https://pr16665-3dba7f732330ad64955bbac009a428331c08d382.ngbuilds.io/

IgorMinar committed:
build(aio): upgrade @angular/* and friends, remove obsolete deps

IgorMinar committed:
build(aio): add ngo + purify

mary-poppins commented:
The angular.io preview for 8b931ea7c3d5d996594a0c08f4ec5cf9ede42f41 is available [here][1].

[1]: https://pr16665-8b931ea7c3d5d996594a0c08f4ec5cf9ede42f41.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 335e21cd3dff931f119f6a3712cd2268322b5259 is available [here][1].

[1]: https://pr16665-335e21cd3dff931f119f6a3712cd2268322b5259.ngbuilds.io/

mary-poppins commented:
The angular.io preview for d938998dc42f9bb80f38d10dbc916ae1ddd8096a is available [here][1].

[1]: https://pr16665-d938998dc42f9bb80f38d10dbc916ae1ddd8096a.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 63827293ab94c9b520348717cd264d97815d921c is available [here][1].

[1]: https://pr16665-63827293ab94c9b520348717cd264d97815d921c.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 3031133cc8102fbab5af923619af42ff613cecbf is available [here][1].

[1]: https://pr16665-3031133cc8102fbab5af923619af42ff613cecbf.ngbuilds.io/

mary-poppins commented:
The angular.io preview for e6ecb90c421852c67dfd318db71e18dff24fd760 is available [here][1].

[1]: https://pr16665-e6ecb90c421852c67dfd318db71e18dff24fd760.ngbuilds.io/

mary-poppins commented:
The angular.io preview for d1f5575ce41e5a2d42e4d08e156feb10cfc464d2 is available [here][1].

[1]: https://pr16665-d1f5575ce41e5a2d42e4d08e156feb10cfc464d2.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 0de6eec7a89a9578923852aff86a510734d0c979 is available [here][1].

[1]: https://pr16665-0de6eec7a89a9578923852aff86a510734d0c979.ngbuilds.io/

gkalpak committed:
fix(aio): fix PWA testing in cases where `atob`/`btoa` is necessary

In some cases (unclear when), traceviewer-js, used by Lighthouse under the hood,
assumes `atob`/`btoa` are defined in the global scope. This is true for browser
environments, but not on node.

As a result, some aggregations that required access to model-tracing failed to
produce results, dropping the overall PWA score.

This affected #16665 (e.g. commit 0de6eec7a).

gkalpak committed:
ci(aio): log the full PWA testing results

This will help diagnose errors or regressions in PWA scores.

gkalpak commented:
https://github.com/angular/angular/pull/16792/commits/d5c41293f1831d7ffa87a1594ffc4d30101f6be3 should fix the PWA testing.

mary-poppins commented:
The angular.io preview for 68ceff8c73443dd685ad91d061a5b2dcf61b0b8a is available [here][1].

[1]: https://pr16665-68ceff8c73443dd685ad91d061a5b2dcf61b0b8a.ngbuilds.io/

IgorMinar commented:
@gkalpak CI still says that PWA scores regressed after I cherry-picked your fixes :-(

IgorMinar committed:
perf(aio): load reflect-metadata polyfill only in the dev mode

mary-poppins commented:
The angular.io preview for 40619aa3b60de1a5965a3041f7bf73b96ec5bdbd is available [here][1].

[1]: https://pr16665-40619aa3b60de1a5965a3041f7bf73b96ec5bdbd.ngbuilds.io/

mary-poppins commented:
The angular.io preview for a0fad2dcae2d3c996898ca3a025a2534ccdcc8af is available [here][1].

[1]: https://pr16665-a0fad2dcae2d3c996898ca3a025a2534ccdcc8af.ngbuilds.io/

IgorMinar committed:
build(aio): upgrade to @angular/cli@1.1.0-rc.0

mary-poppins commented:
The angular.io preview for 4e6f9481b7758e4840e015b6568fca2a96e810fe is available [here][1].

[1]: https://pr16665-4e6f9481b7758e4840e015b6568fca2a96e810fe.ngbuilds.io/

gkalpak committed:
fix(aio): fix PWA testing on Windows (and reduce flaky-ness)

Adding a delay after Lighthouse has run and before killing the Chrome process
avoids `ECONNREFUSED` errors on Windows and will hopefully prevent such errors
occasionally appearing on CI.

Based on info in:
https://github.com/paulirish/pwmetrics/issues/63#issuecomment-282721068

petebacondarwin committed:
fix(aio): fix up typing errors in tests

The new version of Jasmine types identified issues with our unit tests.

mary-poppins commented:
The angular.io preview for c78427298e0672180c78b2aba6c965be3fdc9ff0 is available [here][1].

[1]: https://pr16665-c78427298e0672180c78b2aba6c965be3fdc9ff0.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 6ae4edf836d6542421ae5607f2c66d4acdfee092 is available [here][1].

[1]: https://pr16665-6ae4edf836d6542421ae5607f2c66d4acdfee092.ngbuilds.io/

mary-poppins commented:
The angular.io preview for e86a92332f24ccd8c1acf54b1ccb54b68dff162c is available [here][1].

[1]: https://pr16665-e86a92332f24ccd8c1acf54b1ccb54b68dff162c.ngbuilds.io/

petebacondarwin committed:
build(aio): do not ignore `is-on-https` in lighthouse check

Lighthouse v1.6.5 treats localhost/1.2.7.0.0.1 as secure domains (i.e. as if they where HTTPS), so we need to stop handling the is-on-https audit specially.

mary-poppins commented:
The angular.io preview for 2656db1cbf60128bbc61702c13288ba29102054b is available [here][1].

[1]: https://pr16665-2656db1cbf60128bbc61702c13288ba29102054b.ngbuilds.io/
